src/shared/components/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FunctionComponent, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+	Blankslate,
+	Box,
+	Button,
+	Flex,
+	FlexItem,
+	Icon,
+	IconName,
+	Spacer,
+	Spinner,
+} from '@viaa/avo2-components';
+import { CustomError } from '../../helpers';
+import { AssetType, deleteFile, uploadFile } from '../../services/file-upload-service';
+import toastService from '../../services/toast-service';
+
+import './FileUpload.scss';
+
+interface FileUploadProps {
+	icon?: IconName;
+	label: string;
+	allowedTypes?: string[];
+	assetType: AssetType;
+	ownerId: string;
+	url: string | null;
+	onChange: (url: string | null) => void;
+}
+
+const FileUpload: FunctionComponent<FileUploadProps> = ({
+	icon,
+	label,
+	allowedTypes = ['image/jpeg', 'image/png', 'image/gif'],
+	assetType,
+	ownerId,
+	url,
+	onChange,
+}) => {
+	const [t] = useTranslation();
+	const [isProcessing, setIsProcessing] = useState<boolean>(false);
+
+	const uploadSelectedFile = async (file: File | null) => {
+		if (file) {
+			if (allowedTypes.includes(file.type)) {
+				setIsProcessing(true);
+				const url = await uploadFile(file, assetType, ownerId);
+				onChange(url);
+				setIsProcessing(false);
+			} else {
+				const allowedExtensions = allowedTypes
+					.map(type => type.split('/').pop() || type)
+					.join(', ');
+				toastService.danger(
+					t('Het geselecteerde bestand is niet toegelaten, ({{allowedExtensions}})', {
+						allowedExtensions,
+					})
+				);
+			}
+		}
+	};
+
+	const deleteUploadedFile = async () => {
+		try {
+			setIsProcessing(true);
+			if (url) {
+				await deleteFile(url);
+				onChange(null);
+			}
+		} catch (err) {
+			console.error(new CustomError('Failed to delete asset', err, { url }));
+			toastService.danger(t('Het verwijderen van het bestand is mislukt'));
+		}
+		setIsProcessing(false);
+	};
+
+	// Render
+	return (
+		<Box backgroundColor="gray" className="c-file-upload">
+			{!url ? (
+				<Spacer margin="large">
+					{!isProcessing ? (
+						<>
+							<Flex>
+								{!!icon && (
+									<FlexItem shrink>
+										<Icon size="large" name={icon} />
+									</FlexItem>
+								)}
+								<FlexItem>
+									<Button label={label} ariaLabel={label} type="secondary" autoHeight />
+								</FlexItem>
+							</Flex>
+							<input
+								type="file"
+								onChange={evt =>
```

Comment:
Lijkt me logischer om - indien `evt.target.files` leeg is - gewoon de functie niet uit te voeren in plaats van deze voorwaarde binnen de functie af te handelen.